modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/SearchAsYouTypeFieldMapper.java

```diff
@@ -95,18 +95,14 @@
     public static final TypeParser PARSER
         = new TypeParser((n, c) -> new Builder(n, () -> c.getIndexAnalyzers().getDefaultIndexAnalyzer()));
 
-    private static SearchAsYouTypeFieldMapper toType(FieldMapper in) {
-        return (SearchAsYouTypeFieldMapper) in;
-    }
-
-    private static SearchAsYouTypeFieldType ft(FieldMapper in) {
-        return toType(in).fieldType();
+    private static Builder builder(FieldMapper in) {
+        return ((SearchAsYouTypeFieldMapper)in).builder;
     }
 
     public static class Builder extends ParametrizedFieldMapper.Builder {
 
-        private final Parameter<Boolean> index = Parameter.indexParam(m -> toType(m).index, true);
-        private final Parameter<Boolean> store = Parameter.storeParam(m -> toType(m).store, false);
+        private final Parameter<Boolean> index = Parameter.indexParam(m -> builder(m).index.get(), true);
```

Comment:
These changes are due to position increment handling moving directly into `TextParams.Analyzers`, which simplifies a lot of the palaver around building analyzers for text fields.  SearchAsYouType doesn't actually expose a position increment field so doesn't get the added benefits here, but it still requires a small refactor.